src/main/java/Duke.java

```diff
@@ -12,27 +14,57 @@ public static void main(String[] args) throws IOException {
         System.out.println("Hello from\n" + logo);
 
 
-        String[] arr = new String[100];
-        int curr = 0;
+        ArrayList<Task> arr = new ArrayList<>();
         BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));
 
         String str;
         while ((str = bf.readLine()) != null) {
-            if (str.toLowerCase().equals("bye")) {
+            str = str.toLowerCase();
+
+            if (str.equals("bye")) {
                 System.out.println("Bye. Hope to see you again soon!");
                 break;
             }
 
-            if (str.toLowerCase().equals("list")) {
+            if (str.equals("list")) {
                 int cnt = 1;
-                while (cnt <= curr) {
-                    System.out.println(cnt + ". " + arr[cnt - 1]);
+                while (cnt <= arr.size()) {
+                    Task item = arr.get(cnt - 1);
+                    System.out.println(cnt + ". " + "[" +  item.getStatus() + "] " + item.taskName);
                     cnt++;
                 }
+            } else if (str.length() >= 4 && str.substring(0, 4).equals("mark")) {
```

Comment:
It would be more elegant if this if-else block were to be replaced by a switch statement